[28기 정민아] 리팩토링중입니다

minami-cs opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [ ] 기능 추가
- [x] 리뷰 반영
- [x] 리팩토링
- [x] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- HTML5, SASS, Vanilla JavaScript로 작성한 인스타그램 클론코딩 위스타그램 프로젝트 리액트로 옮기기

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. 로그인 입력 데이터 유효성 검사
- 아이디에 @가 포함되고 비밀번호에 입력한 글자수가 5글자 이상이 되면 유효성 검사를 통과한다.
- 유효성 검사 함수의 반환값(boolean)에 따라 버튼의 배경색과 disabled 속성이 변화한다.
- 테스트 방법 : `/loginminah` 페이지 이동 >>> 로그인 input 데이터 입력

2. 로그인 데이터 처리
- 로그인 페이지에서 입력한 값의 @ 앞부분을 아이디로 세션 스토리지에 저장하여 다른 페이지에서도 사용할 수 있도록 한다.

3. 댓글 데이터 가져오기
- commentData.json에 작성된 댓글 mock data를 가져와서 댓글목록에 출력한다.

4. 댓글 입력 데이터 유효성 검사
- 입력한 글자수가 1이상이 되어 `commentContent`에 들어온 내용이 있을 때 유효성 검사를 통과한다.
- 유효성 검사 통과시 게시 버튼의 글자색과 disabled 속성이 변화한다.
- 테스트 방법 : `/mainminah` 페이지 이동 >>> 댓글 input 데이터 입력

5. 댓글 작성시 enter키 처리
- 댓글 입력 후 enter키를 눌렀을 때 입력한 댓글이 게시되도록 한다.
- 개행처리는 shift키와 enter키를 눌렀을 때 되도록 한다.
- 댓글이 게시될 때에는 개행된 부분에 삽입된 `\n`을 기준으로 댓글을 잘라서 `map`을 이용하여 개행처리를 해준다.

6. 댓글 좋아요 표시
- 각 댓글마다 하트 버튼마다 `isLiked` 속성의 기본값으로 `false`를 부여한다.
- 댓글의 하트 버튼을 누르면 해당 댓글의 `isLiked` 속성이 현재 속성의 반대로 변화한다.
- 댓글의 `isLiked` 속성이 `true`가 되면 붉은색 하트 이미지를 출력하고, `false`일 때에는 속이 빈 하트 이미지를 출력한다.
- 테스트 방법: `/mainminah` 페이지 이동 >>> 댓글의 하트 버튼 클릭

7. 댓글 삭제
- 각 댓글의 X 버튼을 누르면 해당 댓글이 삭제된다.
- X 버튼을 누른 해당 댓글의 아이디를 `filter` 메서드로 걸러서 전체 댓글 목록에서 나타나지 않도록 한다.
- 테스트 방법: `/mainminah` 페이지 이동 >>> 댓글의 X 버튼 클릭

8. 네비게이션바 프로필의 서브메뉴
- 네비게이션바에서 내 프로필 이미지 클릭시 서브메뉴가 나타난다.
- 서브메뉴가 나타났을 때 다른 영역을 클릭하거나 내 프로필 이미지를 클릭하면 서브메뉴가 사라진다.
- 테스트 방법: `/mainminah` 페이지 이동 >>> 네비게이션바 우측의 내 프로필 이미지 클릭

9. 스토리 슬라이드
- 스토리의 왼쪽으로 혹은 오른쪽으로 가기 버튼을 누르면 스토리가 왼쪽 또는 오른쪽으로 슬라이드된다.
- 스토리의 전체 너비, 스토리의 갯수, 스토리 하나의 너비를 구해서 스토리 박스 크기 밖으로 튀어나가서 가려진 부분만큼의 길이를 구하여 translateX로 버튼 클릭시 위치를 X축 방향으로 이동하도록 하고, transform 스타일을 추가하여 이동할 때의 애니메이션이 부드럽게 보이도록 구현

10. 피드 이미지 슬라이드
- 피드 이미지가 2개 이상일 때 왼쪽으로 가기 버튼과 오른쪽으로 가기 버튼이 보이도록 한다.
- 왼쪽으로 혹은 오른쪽으로 가기 버튼을 누르면 다음 또는 이전 이미지가 슬라이드되면서 보이도록 한다.
- 피드의 이미지 갯수, 피드의 너비를 구하고, 인덱스를 부여해서 왼쪽 또는 오른쪽 가기 클릭 시 translateX로 X축 방향으로 이동하도록 하고, transform 스타일을 추가하여 이동할 때 애니메이션이 부드럽게 보이도록 구현


<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
- 댓글목록에 댓글을 추가할 때 두 번 렌더링되어 동일한 댓글이 동시에 두 번 목록에 추가되는 문제를 해결하기 위해 `useCallback()`을 사용했습니다. 상위 컴포넌트에서 하위 컴포넌트로 전달된 `props`가 변할 때 두 번 렌더링되기 때문에 `useCallback()`을 사용하여 두 번 렌더링되지 않도록 할 수 있음을 배웠습니다.
- SASS를 적용하여 만든 SCSS파일을 분리한 컴포넌트를 따라 분리할 경우 단일 SCSS파일에 적용했던 nesting이 똑같이 적용되지 않는 현상을 발견하고 이를 해결하기 위해 분리한 컴포넌트에도 단일 SCSS파일처럼 nesting을 적용할 수 있는 방법을 찾아보았으나 찾지 못하였습니다. 그래서 가장 바깥쪽 태그에 별도의 클래스를 부여하는 것으로 해결하였는데 다른 방법이 있을지 궁금합니다.
- props를 세 번 이상 하위 컴포넌트에 넘겨주는 것을 활용하다 보니 최하위 컴포넌트에서 사용할 때 헷갈리지 않도록 매 컴포넌트 단위마다 props를 잘 넘겨주고 있는지, 또 하위 컴포넌트에서 활용할 때 헷갈리지 않도록 구조 할당을 잘 이용해야 할 것 같습니다.

<br />

## :: 기타 질문 및 특이 사항
- ~아직 `feedData.json` 이름의 mock data만 만들어두고 `commentList`를 제외하고 feed 전체를 불러오는 부분은 구현하지 못했습니다.~ 전체 mock data 활용하도록 수정 완료!

minami-cs committed:
Add: LoginMinah.js와 LoginMinah.scss 추가 및 sass 라이브러리 설치

minami-cs committed:
Merge branch 'master' into feature/minah

minami-cs committed:
Add: LoginMinah.js와 LoginMinah.scss 파일 추가
- LoginMinah.js 파일 추가
- LoginMinah.scss 파일 추가 후 공통 scss파일 임포트 완료

minami-cs committed:
Modify: login 페이지에서 main 페이지로 이동하는 경로 수정
- navigate(/main)에서 navigate(/mainminah)로 수정

minami-cs committed:
Merge branch 'master' into feature/minah

minami-cs committed:
Merge branch 'master' into feature/minah

minami-cs committed:
Add: 로그인 페이지 facebook 아이콘 추가 및 로그인 scss 수정
- Mission 1, Mission 2 완료 및 아이디와 비밀번호 저장 함수명 수정

minami-cs committed:
Add: 로그인 입력폼에서 엔터키를 눌렀을 때 로그인되는 기능 추가

minami-cs committed:
Add: Main 페이지용 각 컴포넌트 생성 및 scss 파일 생성
- Main 페이지 컴포넌트: Aside, Feed, NavigationBar, Story 파일 생성 완료
- Main.scss 생성 완료

minami-cs committed:
Add: 각 컴포넌트별 scss파일 생성
- MainMinah.scss에서 각 컴포넌트별로  scss파일 분리

minami-cs committed:
Add: 피드 댓글작성칸  자동 resize 기능 추가

minami-cs committed:
Add: 엔터키를 누르면 댓글이 작성되도록 하는 기능 구현
- 1글자 이상 댓글 작성시 댓글 게시 버튼 활성화
- 댓글 작성시 shift + enter를 눌렀을 때만 개행하는 로직 추가
- 댓글 작성 기능 구현중

minami-cs committed:
Add: 댓글 mockdata 추가 및 댓글 컴포넌트 수정
- 댓글 mockdata를 commentData.json 파일 이름으로 생성
- 댓글목록을 별도 컴포넌트인 CommentList.js로 분리

minami-cs committed:
Modify: 댓글 mock data를 불러오는 함수 위치 수정
- 댓글 mock data를 불러오는 함수 위치를 최상위 컴포넌트인 main으로 이동

minami-cs committed:
Add: 댓글 추가 기능 구현
- 새 댓글 추가 기능 구현 완료

minami-cs committed:
Modify: 댓글 추가 기능 수정
- 댓글 추가시 textarea에 작성했던 글을 초기화해주도록 수정

minami-cs committed:
Add: 댓글 좋아요 기능 추가
- 댓글의 하트를 클릭시 하트 이미지 변경되는 기능 추가
- 로그인할 때 @ 앞의 아이디만 가져와서 userId에 저장하도록 수정

minami-cs committed:
Add: 댓글 삭제 기능 구현
- 댓글 삭제 기능 구현 완료

minami-cs committed:
Refactor: scss파일에 variables.scss의 변수들을 사용하도록 수정

ghost commented:
Sider has detected **1** warning on analyzing the commit c94694a.

- ⚠️ The number of issues (883) exceeded the limit of comments. Please visit [Sider](https://sider.review/gh/repos/438165292/pulls/2) to review the issues.

If the errors persist even after [retrying](https://sider.review/gh/repos/438165292/pulls/2#tools), the following actions may resolve them:
- Update tool dependencies
- Configure failed tools
- [Configure `sider.yml`](https://sider.review/gh/repos/438165292/pulls/2#config)
- [Turn off unused tools](https://sider.review/gh/repos/438165292/settings)

If you still have problems, feel free to ask us via chat. 💬

---

You can [turn off such notifications](https://sider.review/gh/repos/438165292/settings#notifications) if unnecessary.

minami-cs committed:
Modify: mock data 파일 교체 및 버그 수정
- mock data 파일을 feedData.json으로 교체
- commentList의 길이가 0이 되면 페이지 전체가 표시되지 않게 되어 있던 버그 수정

minami-cs committed:
Fix: eslint 세팅

minami-cs committed:
Refactor: main페이지 aside footer의 a 태그 삭제
- 사용하지 않는 a 태그 삭제

minami-cs committed:
Refactor: console.log 삭제 및 이미지 alt 속성 추가

minami-cs committed:
Refactor: Footer.js의 사용하지 않는 a 태그 삭제

minami-cs committed:
Modify: feedData.json으로 mock data 교체 후 각 컴포넌트 수정
- feedData.json으로 feed의 모든 데이터를 불러오도록 mock data 교체
- 댓글 CRUD 및 좋아요 기능 함수 위치를 메인 페이지에서 Feed 컴포넌트로 이동하고 댓글 컴포넌트별 props 수정
- props를 비구조화 할당하도록 수정

minami-cs committed:
Refactor: 주석 삭제 및 비구조화 할당 적용
- 백엔드와의 API 통신 관련 코드 제외 사용하지 않아 주석 처리한 코드 삭제
- props를 넘겨받은 하위 컴포넌트에 비구조화 할당 모두 적용 완료

minami-cs committed:
Fix: Comment.js에 비구조화할당한 props 이름 때문에 생긴 오류 해결

minami-cs committed:
Modify: 피드 댓글 css 수정
- 댓글 좋아요와 삭제 버튼 위치 수정
- 각 댓글 사이 공간 조금 띄우도록 수정

minami-cs committed:
Refactor: 불필요한 a태그 삭제 및 img 태그의 alt 속성을 맨 앞으로 위치하도록 수정

minami-cs committed:
Modify: westagram 로고 폰트 수정

minami-cs committed:
Add: 스토리 슬라이드 기능 추가

minami-cs committed:
Add: Feed 사진이 여러 개일 때 이미지 슬라이드 기능 추가

minami-cs committed:
Refactor: scss 파일에서 자주 사용되는 변수를 따로 variables 파일에 작성하여 사용하도록 함

minami-cs committed:
Fix: 피드 이미지 슬라이드 기능 인덱스 관련 오류 수정

minami-cs committed:
Refactor: 피드 이미지 슬라이드 코드 리팩토링

minami-cs committed:
Refactor: 로그인시 실행할 함수 분리
- 불필요한 함수 제거
- 백엔드와 통신하는 함수 분리

minami-cs committed:
Fix: 댓글 달기 직후 댓글 입력창 크기 조절이 자동으로 되지 않는 현상 해결
- commentContent와 textarea의 높이를 초기화해주는 것으로 해결

minami-cs committed:
Refactor: 스토리 슬라이드 기능 수정 및 피드 함수 실행부분에 불필요한 코드 삭제

minami-cs committed:
Refactor: 로그인 버튼 활성화부분 코드 줄이기

minami-cs committed:
Refactor: 스토리 이미지 슬라이드 코드에 인덱스를 활용하도록 리팩토링

minami-cs committed:
Refactor: 메인페이지의 json파일 fetch해오는 부분을 함수화한 것을 다시 원래 형태로 수정
